docs/contrib/DatabaseMath.md

```diff
@@ -0,0 +1,90 @@
+# Database Math 101
+
+DEF defines the units it uses with the units command.
+
+```
+    UNITS DISTANCE MICRONS 1000 ;
+```
+
+Typically the units are 1000 or 2000 database units (DBU) per micron.
+DBUs are integers, so the distance resolution is typically 1/1000u or
+1nm.
+
+OpenDB uses an `int` to represent a DBU, which on most hardware is 4
+bytes. This means a database coordinate can be +/-2147483647, which is
+about 2 billion units, corresponding to 2 million microns or 2 meters.
+
+Since chip coordinates cannot be negative, it would make sense to use an
+`unsigned int` to represent a distance. This conveys the fact that it
+can never be negative and doubles the maximum possible distance that can
+be represented. The problem is doing subtraction with unsigned numbers
+is dangerous because the differences can be negative. An unsigned
+negative number looks like a very very big number. So this is a very bad
+idea and leads to bugs.
+
+Note that calculating an area with `int` values is problematic. An
+`int * int` does not fit in an `int`. My suggestion is to use `int64_t`
+in this situation. Although `long` "works", it's size is implementation
+dependent.
+
```

Comment:
References to I, my etc. should be removed- there should be no such personal statements. This whole section needs cleanup.